protocol/x/clob/keeper/msg_server_place_order.go

```diff
@@ -36,76 +36,82 @@ func (k msgServer) PlaceOrder(goCtx context.Context, msg *types.MsgPlaceOrder) (
+		// Attach various logging tags relative to this request. These should be static with no changes.
+		ctx = log.AddPersistentTagsToLogger(ctx,
+			log.Module, log.Clob,
+			log.ProposerConsAddress, sdk.ConsAddress(ctx.BlockHeader().ProposerAddress),
+			log.Callback, lib.TxMode(ctx),
+			log.BlockHeight, ctx.BlockHeight(),
+			log.Handler, log.PlaceOrder,
+			log.Msg, msg,
 		)
-		if err != nil {
-			if errors.Is(err, types.ErrStatefulOrderCollateralizationCheckFailed) {
-				telemetry.IncrCounterWithLabels(
-					[]string{
-						types.ModuleName,
-						metrics.PlaceOrder,
-						metrics.CollateralizationCheckFailed,
-					},
-					1,
-					msg.Order.GetOrderLabels(),
-				)
-				log.InfoLog(ctx, "Place Order Expected Error", log.Error, err)
-				return
+
+		defer func() {
+			metrics.IncrSuccessOrErrorCounter(
+				err,
+				types.ModuleName,
+				metrics.PlaceOrder,
+				metrics.DeliverTx,
+				msg.Order.GetOrderLabels()...,
+			)
+			if err != nil {
+				if errors.Is(err, types.ErrStatefulOrderCollateralizationCheckFailed) {
+					telemetry.IncrCounterWithLabels(
+						[]string{
+							types.ModuleName,
+							metrics.PlaceOrder,
+							metrics.CollateralizationCheckFailed,
+						},
+						1,
+						msg.Order.GetOrderLabels(),
+					)
+					log.InfoLog(ctx, "Place Order Expected Error", log.Error, err)
+					return
+				}
+				log.ErrorLogWithError(ctx, "Error placing order", err)
 			}
-			log.ErrorLogWithError(ctx, "Error placing order", err)
-		}
-	}()
+		}()
+	}
 
 	// 1. Ensure the order is not a Short-Term order.
 	order := msg.GetOrder()
 	order.MustBeStatefulOrder()
 
 	// 2. Return an error if an associated cancellation or removal already exists in the current block.
 	processProposerMatchesEvents := k.GetProcessProposerMatchesEvents(ctx)
-	cancelledOrderIds := lib.UniqueSliceToSet(processProposerMatchesEvents.PlacedStatefulCancellationOrderIds)
-	if _, found := cancelledOrderIds[order.GetOrderId()]; found {
-		return errorsmod.Wrapf(
-			types.ErrStatefulOrderPreviouslyCancelled,
-			"PlaceOrder: order (%+v)",
-			order,
-		)
-	}
-	removedOrderIds := lib.UniqueSliceToSet(processProposerMatchesEvents.RemovedStatefulOrderIds)
-	if _, found := removedOrderIds[order.GetOrderId()]; found {
-		return errorsmod.Wrapf(
-			types.ErrStatefulOrderPreviouslyRemoved,
-			"PlaceOrder: order (%+v)",
-			order,
-		)
+	if !isInternalOrder {
+		cancelledOrderIds := lib.UniqueSliceToSet(processProposerMatchesEvents.PlacedStatefulCancellationOrderIds)
+		if _, found := cancelledOrderIds[order.GetOrderId()]; found {
+			return errorsmod.Wrapf(
+				types.ErrStatefulOrderPreviouslyCancelled,
+				"PlaceOrder: order (%+v)",
+				order,
+			)
+		}
+		removedOrderIds := lib.UniqueSliceToSet(processProposerMatchesEvents.RemovedStatefulOrderIds)
+		if _, found := removedOrderIds[order.GetOrderId()]; found {
+			return errorsmod.Wrapf(
+				types.ErrStatefulOrderPreviouslyRemoved,
+				"PlaceOrder: order (%+v)",
+				order,
+			)
+		}
```

Comment:
on a second thought - this isn't too expensive, right? let's maybe keep this to keep the logic consistent?